integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonSessionState.scala

```diff
@@ -152,6 +152,8 @@ class CarbonSessionState(sparkSession: SparkSession) extends HiveSessionState(sp
         catalog.ParquetConversions ::
         catalog.OrcConversions ::
         CarbonPreInsertionCasts ::
+        // REVIEW_COMMENT: another way is that put this rule after optimization instead of in analysis phase
```

Comment:
column and table resolving is done in analysed plan that is y we are handling here and if any optimization we can apply in optimizer it will be applied that is why it is handled in analysis